sim/pokemon.ts

```diff
@@ -19,29 +19,80 @@ interface MoveSlot {
 }
 
 export class Pokemon {
-	side: Side;
-	battle: Battle;
-	set: PokemonSet;
-	getHealth: (side: Side) => string;
-	getDetails: (side: Side) => string;
+	readonly side: Side;
+	readonly battle: Battle;
+
+	/** "pre-bound" functions for nicer syntax allows them to be passed directly to Battle#add. */
+	readonly getHealth: (side: Side) => string;
+	readonly getDetails: (side: Side) => string;
+
+	readonly set: PokemonSet;
+	readonly name: string;
+	readonly fullname: string;
+	readonly id: string; // shouldn't really be used anywhere
+	readonly species: string;
+	readonly speciesid: string;
+	readonly level: number;
+	readonly gender: GenderName;
+	readonly happiness: number;
+	readonly pokeball: string;
+
+	/** Transform copies IVs in gen 4 and earlier, so we track the base IVs/HP-type/power. */
```

Comment:
Wait, it does? Does it use them for anything other than Hidden Power type?

Reply:
Yes, in general IVs in battle are only used to determine Hidden Power type and power.

Reply:
Let's just copy `hpType`, then.

Reply:
Removed `baseIvs`, kept `hpType` and `hpPower` (I assume you meant for me to keep the latter as well).